Distance_3/include/CGAL/squared_distance_3_0.h

```diff
@@ -218,7 +235,55 @@ squared_distance_to_line(const typename K::Vector_3 & dir,
                               (K*)0, RT(wdot(dir, dir, k)), diff.hw(), diff.hw()));
 }
 
+template <class K>
+void
+squared_distance_to_line_RT(const typename K::Vector_3 & dir,
+                            const typename K::Vector_3 & diff,
+                            typename K::RT& num,
+                            typename K::RT& den,
+                            const K& k,
+                            Cartesian_tag)
+{
+  typedef typename K::Vector_3 Vector_3;
+  typedef typename K::RT RT;
+  typedef typename K::FT FT;
+  Vector_3 wcr = wcross(dir, diff, k);
+  num = wcr*wcr;
+  den = wmult((K*)0, RT(wdot(dir, dir, k)), diff.hw(), diff.hw());
+}
+
+template <class K>
+void
+squared_distance_to_line_RT(const typename K::Vector_3 & dir,
+                            const typename K::Vector_3 & diff,
+                            typename K::RT& num,
+                            typename K::RT& den,
+                            const K& k,
+                            Homogeneous_tag)
+{
+  typedef typename K::Vector_3 Vector_3;
+  typedef typename K::RT RT;
+  typedef typename K::FT FT;
+  Vector_3 wcr = wcross(dir, diff, k);
+  FT ft = FT(wcr*wcr)/FT(wmult(
+                               (K*)0, RT(wdot(dir, dir, k)), diff.hw(), diff.hw()));
+  num = Rational_traits<FT>().numerator(ft);
+  den = Rational_traits<FT>().denominator(ft);
+}
```

Comment:
I am a bit worried by the code duplication between the functions `squared_distance_to_something` and `squared_distance_to_something_RT`.

At least for the homogeneous kernel, one function can be reimplemented easily by the other:
```suggestion
template <class K>
void
squared_distance_to_line_RT(const typename K::Vector_3 & dir,
                            const typename K::Vector_3 & diff,
                            typename K::RT& num,
                            typename K::RT& den,
                            const K& k,
                            Homogeneous_tag)
{
  typedef typename K::FT FT;
  FT ft = squared_distance_to_line(dir, diff, k);
  num = Rational_traits<FT>().numerator(ft);
  den = Rational_traits<FT>().denominator(ft);
}
```

Reply:
looks like a recursion.

Reply:
`squared_distance_to_line_RT` with `Homogeneous_tag` as the last argument would call `squared_distance_to_line`. That is not a recursion, because the code of `squared_distance_to_line` is:
https://github.com/CGAL/cgal/blob/b7b2e1955556defdb60088db16ab7d7404c228c7/Distance_3/include/CGAL/squared_distance_3_0.h#L224-L236

Reply:
Why not go the other way, with:
```
FT f(p...)
{
  	RT num, denum;
    f_RT(p, num, denum)
    return num/denum;
}
```
?

Reply:
If this involves homogeneous where RT!=FT, num/den is probably Fraction_traits::Compose()(num,den) (I don't remember if there is a short notation for it).